java/arcs/core/data/expression/Evaluator.kt

```diff
@@ -46,15 +49,25 @@ class ExpressionEvaluator(val parameterScope: Expression.Scope = ParameterScope(
 
     override fun visit(expr: Expression.BooleanLiteralExpression): Boolean = expr.value
 
+    override fun <T : Expression.Scope> visit(expr: Expression.CurrentScopeExpression<T>) =
+        currentScope
+
     override fun <T> visit(expr: Expression.ObjectLiteralExpression<T>): Any = expr.value as Any
 }
 
 /**
  * Given an expression, and a list of parameter mappings, evaluate the expression and return
  * the result using an [ExpressionEvaluator].
+ * @param expression the expression to be evaluated
+ * @param currentScope a [Scope] object for lookups in [Expression.CurrentScopeExpression]
+ * @param params mappings of query-args by name for [Expression.QueryParameterExpression]
  */
-fun <T, R> evalExpression(expression: Expression<T>, vararg params: Pair<String, Any>): R {
+fun <T, R> evalExpression(
+    expression: Expression<T>,
+    currentScope: Expression.Scope = mapOf<String, Any>().asScope(),
+    vararg params: Pair<String, Any>
```

Comment:
Food for thought - maybe we could unify currentScope and parameterScope here? WDYT?

Reply:
I think `currentScope` is for accessing into entities, which should be tracked separate to the `parameterScope`.
Merging them might make the following ambiguous

`Foo {a: inline {x: Number}, x: Number} [a.x + x > 100]`

Reply:
parameterScope is for queryArgs (?arg) resolution.